src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/interceptor/CustomUserAuthenticationInterceptor.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.interceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.util.security.AuthorisationUtil;
+import uk.gov.companieshouse.api.util.security.Permission;
+import uk.gov.companieshouse.api.util.security.SecurityConstants;
+import uk.gov.companieshouse.limitedpartnershipsapi.utils.ApiLogger;
+
+import java.util.HashMap;
+
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+@Component
+public class CustomUserAuthenticationInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull Object handler) {
+        String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
+
+        if (skipTokenPermissionChecksWhenApiKeyUsed(reqId, request)) {
+            return true;
+        }
+
+        // TokenPermissions should have been set up in the request by TokenPermissionsInterceptor
+        final var tokenPermissions = AuthorisationUtil.getTokenPermissions(request)
+                .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
```

Comment:
exception message references UserAuthenticationInterceptor instead of CustomUserAuthenticationInterceptor

Reply:
Updated exception message.